http://www.openoffice.org/issues/show_bug.cgi?id=39958
http://www.openoffice.org/issues/show_bug.cgi?id=38417

o Not for integration, we should provide configure based one
  Note: malloc.h is deprecated for FreeBSD.
o use configure to produce config.h

Index: libxmlsec/xmlsec1-1.2.4.patch
===================================================================
RCS file: /cvs/external/libxmlsec/xmlsec1-1.2.4.patch,v
retrieving revision 1.7
diff -u -r1.7 xmlsec1-1.2.4.patch
--- libxmlsec/xmlsec1-1.2.4.patch	9 Sep 2004 08:31:29 -0000	1.7
+++ libxmlsec/xmlsec1-1.2.4.patch	3 Oct 2004 05:52:20 -0000
@@ -83,7 +83,7 @@
 ! #undef HAVE_MACH_O_DYLD_H
 ! 
 ! /* Define to 1 if you have the <malloc.h> header file. */
-! #define HAVE_MALLOC_H 1
+! #define HAVE_MALLOC_H 0
 ! 
 ! /* Define to 1 if you have the `memcpy' function. */
 ! #undef HAVE_MEMCPY


Index: xmlsecurity/source/xmlsec/xmldocumentwrapper_xmlsecimpl.cxx
===================================================================
RCS file: /cvs/xml/xmlsecurity/source/xmlsec/xmldocumentwrapper_xmlsecimpl.cxx,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 xmldocumentwrapper_xmlsecimpl.cxx
--- xmlsecurity/source/xmlsec/xmldocumentwrapper_xmlsecimpl.cxx	12 Jul 2004 13:15:21 -0000	1.1.1.1
+++ xmlsecurity/source/xmlsec/xmldocumentwrapper_xmlsecimpl.cxx	4 Oct 2004 22:04:08 -0000
@@ -68,7 +68,9 @@
 #include <xmloff/attrlist.hxx>
 #include "xmlelementwrapper_xmlsecimpl.hxx"
 
+#if !defined (FREEBSD)
 #include <malloc.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
